### PR TITLE
#3046 Error message for temperature syncing

### LINKED ISSUE #3046: Show a more detailed error message on bluetooth syncing errors
## Is your feature request related to a problem? Please describe.

Currently a message displays "Error downloading logs" for example, but it is not more detailed. To more easily diagnose issues, display a more detailed error message.

## Describe the solution you'd like

Currently messages from the bluetooth module are sent with the error. Display these.

## Implementation

Take the error code in the error and set it in the store. Use that to lookup a translated string or the error message.

## Describe alternatives you've considered

N/A

## Additional context

Alain can't sync temperatures, I can. I don't know what's wrong - trying to get more details to help him get it working!

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3046 

## Change summary

- Adds a more detailed error message when syncing

## Testing

Alain! 🤣 

### Related areas to think about

I doubt this will fix @alainsussol 's problem. I imagine it is to do with the device he's on. Bluetooth, especially on android is pretty horrific. Well, actually using most native features is pretty annoying given the number of different OS'/devices! But specifically he is on a samsung and they like to be difficult!

### CODE REVIEW

File: android/app/src/main/java/com/msupplymobile/Ble/ErrorCode.java [hunk 1]
```diff
@@ -11,8 +11,8 @@
     E_INVALID_ADDRESS("Problem setting the address filter"),
     E_SCAN_FAILED("Problem occurred while scanning"),
     E_NULL_SCAN_RESULT("A scan resulted in a null value"),
-    E_UNKOWN_SCAN_RESULT("An unkown scan result was retrieved"),
-    E_CANT_SCAN("Problem when iniating a BLE scan"),
+    E_UNKNOWN_SCAN_RESULT("An unknown scan result was retrieved"),
+    E_CANT_SCAN("Problem when initiating a BLE scan"),
```
Comment (joshxg) [inline android/app/src/main/java/com/msupplymobile/Ble/ErrorCode.java:15]: these are sent in a message from the native module

File: src/localization/vaccineStrings.json [hunk 2]
```diff
@@ -10,7 +10,22 @@
...
+    "E_IN_DESC_WRITE": "Problem writing an incoming descriptor",
+    "E_OUT_CHAR_WRITE": "Problem writing an outgoing characteristic",
+    "E_INVALID_ADDRESS": "Problem setting the address filter",
+    "E_SCAN_FAILED": "Problem occurred while scanning",
+    "E_NULL_SCAN_RESULT": "A scan resulted in a null value",
+    "E_UNKNOWN_SCAN_RESULT": "An unknown scan result was retrieved",
+    "E_CANT_SCAN": "Problem when initiating a BLE scan",
+    "E_BLUETOOTH_DISABLED": "Problem enabling the bluetooth adapter",
+    "E_BLUETOOTH_ADAPTER": "Problem accessing the bluetooth adapter",
+    "E_BLUETOOTH_DENIED": "Denied access to the bluetooth adapter"
```
Comment (joshxg) [inline src/localization/vaccineStrings.json:28]: maybe lazy? I dunno if these are the best keys 🤷

File: src/reducers/TemperatureSyncReducer.js [hunk 3]
```diff
@@ -32,6 +32,7 @@ const initialState = () => ({
   syncError: '',
   currentSensorName: null,
   isDisabled: false,
+  errorMessage: '',
```
Comment (joshxg) [inline src/reducers/TemperatureSyncReducer.js]: clear/set the erorr code

**Comment by joshxg** [comment]
Thanks @mark-prins 

Wish I could make testing easier for you :(

**Comment by mark-prins** [comment]
I take it all back - I have this branch running on an actual tablet, with sensors and fridges configured. all ready to test the next PR that comes along!

**Comment by joshxg** [comment]
@mark-prins 🎉 🎉 🎉 

im so excited

**Pull request merged**